### PR TITLE
Groovy library for validation workflow

### PR DESCRIPTION
**Pull request by Divyaasm**
Signed off by 
### Description
Created a groovy library for Validation Workflow

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: vars/validateArtifacts.groovy [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/**
+SignArtifacts signs the given artifacts and saves the signature in the same directory.
```
Comment (gaiksaya) [inline vars/validateArtifacts.groovy:11]: Description needs to be fixed!

File: vars/validateArtifacts.groovy [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@param Map[version] <Required> - Version of the opensearch artifact that needs to be validated.
+@param Map[distribution] <Optional> - Choose distribution type among [tar(default),rpm,yum,docker].
+@param Map[architecture] <Optional> - Select the architecture type among [x64(default) and arm64].
+@param Map[platform] <Optional> - The distribution platform.
+@param Map[docker_source] <Optional> - Specify the docker source from [DockerHub(default), ECR] to pull the docker image.
+@param Map[os_build_number]<Optional> - Specify Opensearch build number from opensearchstaging if required.
+@param Map[osd_build_number]<Optional> - Specify Opensearch-Dashboard build number from opensearchstaging if required.
```
Comment (gaiksaya) [inline vars/validateArtifacts.groovy:18]: You can remove this and say pass whatever needs to be passed to validation script. Add more details how they are passed. example `--<key> <value>`

**Review by gaiksaya — CHANGES_REQUESTED** [review]
Need to add tests and libtester. See details here: https://github.com/opensearch-project/opensearch-build-libraries/blob/main/DEVELOPER_GUIDE.md#tests-for-jenkins-libraries

File: vars/validateArtifacts.groovy (hunk 2 above)
Comment (Divyaasm) [reply vars/validateArtifacts.groovy:18]: Sure @gaiksaya , can I completely remove the above lines and specify in detail how each argument is passed to the workflow with a help statement to understand how to run the shell script?

File: vars/validateArtifacts.groovy (hunk 1 above)
Comment (Divyaasm) [reply vars/validateArtifacts.groovy:11]: My bad, missed it while shifting between branches, will update it, thanks

**Comment by gaiksaya** [comment]
Closing in favor of https://github.com/opensearch-project/opensearch-build-libraries/pull/173

**Pull request closed without merging**